core/server/data/migrations/versions/3.4/01-add-self-signup-and-from address-to-members-settings.js

```diff
@@ -0,0 +1,107 @@
+const _ = require('lodash');
+
+const common = require('../../../../lib/common');
+
+module.exports.config = {
+    transaction: true
+};
+
+const settingsKey = 'members_subscription_settings';
+const complimentaryPlan = {
+    name: 'Complimentary',
+    currency: 'usd',
+    interval: 'year',
+    amount: '0'
+};
+
+module.exports.up = (options) => {
+    let localOptions = _.merge({
+        context: {internal: true}
+    }, options);
+
+    return localOptions
+        .transacting('settings')
+        .then((response) => {
+            if (!response) {
+                common.logging.warn('Cannot find settings.');
+                return;
+            }
+
+            let subscriptionSettingsEntry = response.find((entry) => {
+                return entry.key === settingsKey;
+            });
+
+            if (!subscriptionSettingsEntry) {
+                common.logging.warn('Cannot find members subscription settings.');
+                return;
+            }
+
+            let subscriptionSettings = JSON.parse(subscriptionSettingsEntry.value);
+
+            let paymentProcessors = subscriptionSettings.paymentProcessors;
+
+            if (paymentProcessors && paymentProcessors[0].adapter === 'stripe') {
+                common.logging.debug(`Currently defined plans: ${JSON.stringify(paymentProcessors[0].config.plans)}`);
+                let stripePlans = paymentProcessors[0].config.plans;
+                const hasComplimentaryPlan = stripePlans.find(plan => (plan.name === complimentaryPlan.name));
+
+                if (!hasComplimentaryPlan) {
+                    common.logging.info('Adding "Complimentary" plan to stripe adapter');
+                    stripePlans.push(complimentaryPlan);
+                }
```

Comment:
@gargol Migration looks 👍, only extra addition we can do here is a negative check which we usually add to migrations for better migration logging - i.e. if there is already a complimentary plan(edge case) do `logging.warn('Complimentary plan already exists')` and similarly in the `down` migration if it doesn't exist already do `logging.warn('No complimentary plan found')`, if that makes sense. 😄

Reply:
Yes, these two additional logs won't hurt :ok_hand: will add them. Thanks for the suggestion!